source/common/http/conn_manager_impl.cc

```diff
@@ -37,6 +37,31 @@
 namespace Envoy {
 namespace Http {
 
+namespace {
+
+template <class T> using FilterList = std::list<std::unique_ptr<T>>;
+
+// Shared helper for recording the latest filter used.
+template <class T>
+void recordLatestDataFilter(const typename FilterList<T>::iterator current_filter,
+                            T** latest_filter, const FilterList<T>& filters) {
```

Comment:
nit: small preference for passing a reference to a pointer, since it can't be nullptr, and less dereferencing below, but up to you.